jupyter_book/book_template/_includes/sidebar.html

```diff
@@ -35,17 +35,29 @@ <h2 class="c-sidebar__title">{{ site.title }}</h2>
       Find the title for this chapter
       {% endcomment %}
       {% assign chapter_url_html = chapter.url | append: '.html' %}
-      {% for build_page in site.build %}
-        {% if build_page.url == chapter_url_html %}
-          {% assign chapter_title = build_page.title %}
+      {% if chapter.anchor %}
+        {% if chapter.title %}
+          {% assign chapter_title = chapter.title %}
+        {% else %}
+          {% assign chapter_title = chapter.anchor | replace: "-", " " %}
```

Comment:
I'm not sure - I think the names for the anchor links are determined by `anchor.js`, which I'm not familiar with. I don't know if the HTML-ifying pattern that `anchor.js` uses is the default or a standard or whether it will change in the future.